UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -80,6 +80,12 @@ public struct AgentInfo
         /// </summary>
         public int id;
 
+        /// <summary>
+        /// User-customizable object for sending structured output from Unity to Python in response
+        /// to an action in addition to a scalar reward.
+        /// </summary>
+        public CustomObservation customObservation;
```

Comment:
Why does this field have to be public ?

Reply:
I made it public since all the other properties in `AgentInfo`, where this declaration resides, are public. `Agent` directly accesses the field in `SetCustomObservation` and `ResetData`. The `AgentInfo` instance attached to the agent is still private though to the agent, though.